src/main/webapp/js/user-page-loader.js

```diff
@@ -129,11 +154,41 @@ function fetchMessages() {
       });
 }
 
-/** Fetches replies and adds them to their parent message. */
+/**
+ * Fetches replies and adds them to their parent message.
+ * @param {Message} message
+ */
 function fetchReplies(message) {
+
+  //create reply thread
   const replyThread = document.createElement('div');
   replyThread.classList.add('reply-thread');
+  replyThread.classList.add('hidden');
+  replyThread.id = guidGenerator();
+  var threadID = replyThread.id;
+
+  //create reply toggles
+  const replyToggle = document.createElement('button');
+  replyToggle.classList.add('reply-toggle');
+  replyToggle.id = guidGenerator();
+  var toggleID = replyToggle.id;
+  replyToggle.classList.add('hidden');
+  replyToggle.innerHTML = "Show Replies";
+
+  replyToggle.addEventListener("click", function() {
+    toggleReplies(threadID, toggleID, replyToggle.innerHTML);
```

Comment:
Instead of generating a random ID for the `threadID` and `toggleID`, I'm wondering if you could pass the elements in directly, so pass `replyThread` and `replyToggle` into the `toggleReplies` function as arguments, that way you don't have to look them up again (and you don't have to generate IDs at all)?

Reply:
My first attempts at that have failed, so I'll keep the current code for now.

Reply:
As for loading the replies later, I've also been debating on loading the thread on a whole new page. I've seen some projects in the showcase do that. I might consider switching to one of these approaches at some point.